csrc/dynamic_transform.cpp

```diff
@@ -325,10 +326,18 @@ void DynamicTransformConcretizationInfo::analyzeReshapes(
           "Invalid evaluated value of domain extent: ",
           inp_id->toString());
       NVF_ERROR(
-          extent_val.as<int64_t>() > 0,
+          extent_val.as<int64_t>() >= 0,
           "Invalid input domain extent: ",
           extent_val.as<int64_t>());
       inp_shape.at(i) = extent_val.as<int64_t>();
+      if (inp_shape.at(i) == 0l) {
```

Comment:
I understand you are trying to special case size-0 dimensions, which is a valid solution and a strict improvement. What breaks if we don't do that and just change the check at line 329 to `>=`? Usually, if implemented well, size-0 can be handled elegantly as a general case.

Reply:
Yeah that is where I started. There are some problems with ignoring size 0. For one, when we pass a -1 shape and the input has zero size, we need to insert a 0 in that output position. Then, `analyzeView` does not properly handle the case where the input is empty. In such case there are no transforms to be performed (you can reshape an empty tensor to any empty shape). The `AnalyzeViewResult` struct only holds that transform info, and it cannot refer to `Val`s. That means we have no way to actually determine the output size based only on the `AnalyzeViewResult` alone. Supposing we could create the `FullOp` output, we then need to concretize its `IterDomain`s since the extent expressions might not have previously appeared. We also need to make sure the remove-empty pass can pick up the result properly, e.g. to translate reductions of empty tensors into full ops themselves. When you start pulling one thread it tends to snowball! But I am pretty close to getting this working.